Data/Vector/Generic.hs

```diff
@@ -2179,3 +2183,24 @@ dataCast :: (Vector v a, Data a, Typeable1 v, Typeable1 t)
          => (forall d. Data  d => c (t d)) -> Maybe  (c (v a))
 {-# INLINE dataCast #-}
 dataCast f = gcast1 f
+
+{-# INLINE groupBy #-}
+groupBy k xs =
+  switchL []
+  (\h t ->
+      let n = 1 + findIndexOrEnd (not . k h) t
+      in  unsafeTake n xs : groupBy k (unsafeDrop n xs))
+  xs
+
+-- | 'findIndexOrEnd' is a variant of findIndex, that returns the length
+-- of the string if no element is found, rather than Nothing.
```

Comment:
Is it faster than `fromMaybe len (findIndex p)`?
I think it is better to pattern-match on result of `findIndex` in `groupBy`: if it's `Nothing` you do not need to make another recursive call.